### PR TITLE
Update xbox controller control option

### PR DESCRIPTION
**Pull request by wenjie-mo**
## What changes do you make in this PR?
Add Xbox controller control (both wired and wireless) for Metadrive

## Checklist

* [x] I have merged the latest main branch into current branch.
* [x] I have run `bash scripts/format.sh` before merging.
* Please use "squash and merge" mode.

### CODE REVIEW

File: metadrive/engine/core/manual_controller.py [hunk 1]
```diff
@@ -185,3 +186,75 @@ def process_input(self, vehicle):
...
+        self.button_a = True if self.joystick.get_button(self.BUTTON_A_MAP) else False
+        self.button_b = True if self.joystick.get_button(self.BUTTON_B_MAP) else False
+
+        hat = self.joystick.get_hat(0)
+        self.button_up = True if hat[-1] == 1 else False
+        self.button_down = True if hat[-1] == -1 else False
+        self.button_left = True if hat[0] == -1 else False
+        self.button_right = True if hat[0] == 1 else False
+
+        return [steering, throttle_brake / 2]
```
Comment (pengzhenghao) [inline metadrive/engine/core/manual_controller.py:260]: why divided by 2?

File: metadrive/policy/manual_control_policy.py [hunk 2]
```diff
@@ -25,9 +25,12 @@ def __init__(self, obj, seed):
         if config["manual_control"]:
             if config["controller"] == "keyboard":
                 self.controller = KeyboardController(pygame_control=pygame_control)
-            elif config["controller"] == "joystick":
+            elif config["controller"] in ["joystick", "xboxController"]:
                 try:
-                    self.controller = SteeringWheelController()
+                    if config["controller"] == "joystick":
+                        self.controller = SteeringWheelController()
+                    elif config["controller"] == "xboxController":
```
Comment (pengzhenghao) [inline metadrive/policy/manual_control_policy.py:32]: "xboxController" might cause error when typing, try use ["xboxController", "xboxcontroller", "xbox"] and more options that the user might want to use

Comment (wenjie-mo) [reply metadrive/policy/manual_control_policy.py]: updated

File: metadrive/engine/core/manual_controller.py (hunk 1 above)
Comment (wenjie-mo) [reply metadrive/engine/core/manual_controller.py]: Divide by 2 is not needed, removed.

**Review by QuanyiLi — APPROVED** [review]
LGTM

**Pull request merged**